drivers/sdhc/imx_usdhc.c

```diff
@@ -531,6 +544,70 @@ static int imx_usdhc_card_busy(const struct device *dev)
 		       : 1;
 }
 
+#ifdef CONFIG_SDHC_SCATTER_GATHER_TRANSFER
+/* Convert normal usdhc_data_t to usdhc_scatter_gather_data_t */
+static void imx_usdhc_to_sg_data(usdhc_scatter_gather_data_t *sg_data, usdhc_data_t *data)
+{
+	sg_data->enableAutoCommand12 = data->enableAutoCommand12;
+	sg_data->enableAutoCommand23 = data->enableAutoCommand23;
+	sg_data->enableIgnoreError = data->enableIgnoreError;
+	sg_data->dataType = data->dataType;
+	sg_data->blockSize = data->blockSize;
+
+	if (data->rxData != NULL) {
+		sg_data->dataDirection = kUSDHC_TransferDirectionReceive;
+		sg_data->sgData.dataAddr = data->rxData;
+	} else {
+		sg_data->dataDirection = kUSDHC_TransferDirectionSend;
+		sg_data->sgData.dataAddr = (uint32_t *)data->txData;
+	}
+
+	sg_data->sgData.dataSize = data->blockSize * data->blockCount;
+	sg_data->sgData.dataList = NULL;
+}
+
+static void imx_usdhc_free_sg_list(usdhc_scatter_gather_data_t *usdhc_sg_data)
+{
+	usdhc_scatter_gather_data_list_t *current = usdhc_sg_data->sgData.dataList;
+
+	while (current != NULL) {
+		usdhc_scatter_gather_data_list_t *next = current->dataList;
+
+		k_free(current);
+		current = next;
+	}
+}
+
+static int imx_usdhc_fill_sg_list(usdhc_scatter_gather_data_t *usdhc_sg_data,
+				  struct net_buf *sg_data)
+{
+	usdhc_scatter_gather_data_list_t *usg = &(usdhc_sg_data->sgData);
+	struct net_buf *sg = sg_data;
+
+	while (true) {
+		usg->dataAddr = (uint32_t *)sg->data;
+		usg->dataSize = sg->len;
+		if (sg->frags == NULL) {
+			usg->dataList = NULL;
+			return 0;
+		}
+
+		usg->dataList = k_malloc(sizeof(usdhc_scatter_gather_data_list_t));
```

Comment:
Kind of concerned by the use of `k_malloc` here. Can be allocate these entries statically?

Reply:
Okay. static allocation will be easier to manage, but user may need to configure enough the buffer count to use.
Let me update the PR.
Thanks.